### PR TITLE
Fix of deprecated warnings Julia 0.4.0

### PR DESCRIPTION
**Pull request by RobertBeyer**

### CODE REVIEW

File: src/MAT.jl [hunk 1]
```diff
@@ -32,8 +32,8 @@ using HDF5, MAT_HDF5, MAT_v5
 export matopen, matread, matwrite, names, exists, @read, @write
 
 # Open a MATLAB file
-const HDF5_HEADER = Uint8[0x89, 0x48, 0x44, 0x46, 0x0d, 0x0a, 0x1a, 0x0a]
-function matopen(filename::String, rd::Bool, wr::Bool, cr::Bool, tr::Bool, ff::Bool)
```
Comment (simonster) [inline src/MAT.jl]: Should be `AbstractString`, not `AbstractUInt8`

**Comment by simonster** [comment]
Thanks for doing this! You need to change `using HDF5, MAT_HDF5, MAT_v5` at the top of the file `using HDF5, MAT_HDF5, MAT_v5, Compat` to avoid breaking on 0.3. Also, it's usually more convenient if you combine multiple related changes into a single pull request.

**Comment by simonster** [comment]
Closed by #48

**Comment by RobertBeyer** [comment]
Hi,
I wanted to bundle the changes, but I didn't find the right button to click on here...
Anyway, thanks for fixing the issue properly :+1:

**Comment by simonster** [comment]
Thanks for your efforts! It's actually @phobon not me who deserves the credit here.

The GitHub interface is fine for simple changes, but if you want to modify multiple files or change existing pull requests, you're better off with command-line git, or maybe [GitHub Desktop](https://desktop.github.com/).

**Pull request closed without merging**